uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Test/Fabricator/Task.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * Assignments Fabricator class
+ */
+class CRM_Tasksassignments_Test_Fabricator_Task {
+  private static $defaultParams = [
+    'activity_name' => 'Test Task',
+    'source_contact_id' => 'user_contact_id',
+    'target_id' => 'user_contact_id',
+    'assignee_id' => 'user_contact_id',
+  ];
+
+  /**
+   * Fabricates an assignment merging given parameters with default minimum
+   * parameters.
+   *
+   * @param array $params
+   *   List of parameters to use to create the Assignment
+   *
+   * @return array
+   */
+  public static function fabricate($params = []) {
+    $params = array_merge($params, self::$defaultParams);
+    $result = civicrm_api3('Task', 'create', $params);
+
+    return array_shift($result['values']);
```

Comment:
Given that the other fabricators in this extension (`Document` and `Assignment`) use the the BAO instead of the API, I think we should do the same here

Reply:
It is doing something different alright. Now we have fabricators in CiviHR, some of which return and array, some of which return a BAO. In T&A they return a BAO. Before I change this can we agree on an interface for fabricators so we don't have to check them each time. I know you didn't like my suggestion to have a FabricatorInterface, but I can tell you after working with them for the last few days it would have helped me a lot. Not just for autocompletion (although there were times that would have been useful) but just so I can know what parameters are required and what the return type is.

Reply:
Let's open this discussion to other devs and see what they think, but my opinion is: For anything that is not a core entity, we should use BAOs instead of APIs, for the single reason that API calls are very slow and hurt the performance of the tests. There might be exceptions to this, as I know that some extensions have logic implemented directly at the API level (HRJobContract for example), and in that case there's no other option.

For now, let's use the BAO here

Reply:
After discussion with Davi the decision was to continue using API calls for T&A entities as it can be difficult to find what params to pass to BAO calls for T&A.